common/include/pcl/pcl_macros.h

```diff
@@ -370,3 +370,29 @@ aligned_free (void* ptr)
 #else
   #define PCL_NODISCARD
 #endif
+
+/**
+ * \brief Macro choose between TYPED_TEST_CASE and PCL_TYPED_TEST_SUITE depending on the GTest version
+ *
+ * \ingroup common
+ */
+#if defined(TYPED_TEST_SUITE)
+  #define PCL_TYPED_TEST_SUITE TYPED_TEST_SUITE
+#elif defined(TYPED_TEST_CASE)
+  #define PCL_TYPED_TEST_SUITE TYPED_TEST_CASE
+#else
+  #define PCL_TYPED_TEST_SUITE PLEASE_INCLUDE_GTEST_H_BEFORE_PCL_MACROS_H
+#endif
```

Comment:
I forgot to mention, that I would rewrite this section (and the next one) as following
```cpp
#if !defined(TYPED_TEST_SUITE)
  #define TYPED_TEST_SUITE TYPED_TEST_CASE
#endif
```

If the user is using an old GTest version, we define the new symbol for him/her/(your preferred pronoun).
And then convert all current uses of `TYPED_TEST_CASE` to the new syntax `TYPED_TEST_SUITE`

Reply:
Done.